dist/tools/openocd/openocd.sh

```diff
@@ -0,0 +1,177 @@
+#!/bin/sh
+#
+# Unified OpenOCD script for RIOT
+#
+# This script is supposed to be called from RIOTs make system,
+# as it depends on certain environment variables. An OpenOCD
+# configuration file must be present in a the boards dist folder.
+#
+# The script supports the following actions:
+#
+# flash:        flash a given hexfile to the target.
+#               hexfile is expected in ihex format and is pointed to
+#               by HEXFILE environment variable
+#
+#               options:
+#               HEXFILE: path to the hexfile that is flashed
+#
+# debug:        starts OpenOCD as GDB server in the background and
+#               connects to the server with the GDB client specified by
+#               the board (DBG environment variable)
+#
+#               options:
+#               GDB_PORT:       port opened for GDB connections
+#               TCL_PORT:       port opened for TCL connections
+#               TELNET_PORT:    port opened for telnet connections
+# 		TUI: 		if TUI!=null, the -tui option will be used
+#
+# debug-server: starts OpenOCD as GDB server, but does not connect to
+#               to it with any frontend. This might be useful when using
+#               IDEs.
+#
+#               options:
+#               GDB_PORT:       port opened for GDB connections
+#               TCL_PORT:       port opened for TCL connections
+#               TELNET_PORT:    port opened for telnet connections
+#
+# reset:        triggers a hardware reset of the target board
+#
+#
+# @author       Hauke Peteresen <hauke.petersen@fu-berlin.de>
+
+# default GDB port
+_GDB_PORT=3333
+# default telnet port
+_TELNET_PORT=4444
+# default TCL port
+_TCL_PORT=6333
+# path to OpenOCD configuration file
+CONFIG=${RIOTBOARD}/${BOARD}/dist/openocd.cfg
+
+#
+# a couple of tests for certain configuration options
+#
+test_config() {
+    if [ ! -f ${CONFIG} ]; then
+        echo "Error: Unable to locate OpenOCD configuration file"
+        echo "       (${CONFIG})"
+        exit 1
+    fi
+}
+
+test_hexfile() {
+    if [ ! -f ${HEXFILE} ]; then
+        echo "Error: Unable to locate HEXFILE"
+        echo "       (${HEXFILE})"
+        exit 1
+    fi
+}
+
+test_ports() {
+    if [ -z ${GDB_PORT} ]; then
+        GDB_PORT=${_GDB_PORT}
+    fi
+    if [ -z ${TELNET_PORT} ]; then
+        TELNET_PORT=${_TELNET_PORT}
+    fi
+    if [ -z ${TCL_PORT} ]; then
+        TCL_PORT=${_TCL_PORT}
+    fi
+}
+
+test_tui() {
+    if [ ! -z ${TUI} ]; then
+        TUI=-tui
```

Comment:
Ah, I did not realize there was a `!` in front of the `-z`, which made me read: If there is no `-tui` set, make it so.

Maybe it would be better to use the `-n` (the length of STRING is nonzero) flag of `test` instead of `-z` (the length of STRING is zero) so this is clearer. Additionally the `test` would have less to parse ;-)